lib/rubocop/cop/rails/pluralization_grammar.rb

```diff
@@ -0,0 +1,93 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for correct grammar when using active support's
+      # core extensions to the numeric classes
+      #
+      # @example
+      #   # bad
+      #   3.day.ago
+      #
+      #   # good
+      #   45.minutes.from_now
+      class PluralizationGrammar < Cop
+        DURATION_METHODS = { second: :seconds,
+                             minute: :minutes,
+                             hour: :hours,
+                             day: :days,
+                             week: :weeks,
+                             fortnight: :fortnights,
+                             month: :months,
+                             year: :years }
+
+        MSG = 'Prefer %s.%s for grammar reasons.'
+
+        def on_send(node)
+          receiver, method_name, *_args = *node
+          return if receiver.nil?
+          return unless duration_method?(method_name)
+          return unless literal_number?(receiver)
+          number, = *receiver
+          if singular_receiver?(number) && plural_method?(method_name)
+            add_offense(node,
+                        :expression,
+                        format(MSG, number, singularize(method_name)))
+          elsif plural_receiver?(number) && singular_method?(method_name)
+            add_offense(node,
+                        :expression,
+                        format(MSG, number, pluralize(method_name)))
+          end
+        end
+
+        private
+
+        def autocorrect(node)
+          lambda do |corrector|
+            method_name = node.loc.selector.source
+            replacement = if plural_method?(method_name)
+                            singularize(method_name).to_s
+                          else
+                            pluralize(method_name).to_s
+                          end
+            corrector.replace(node.loc.selector, replacement)
+          end
+        end
+
+        def plural_method?(method_name)
+          method_name.to_s.end_with?('s')
+        end
+
+        def singular_method?(method_name)
+          !plural_method?(method_name)
+        end
+
+        def singular_receiver?(number)
+          number == 1
+        end
+
+        def plural_receiver?(number)
+          !singular_receiver?(number)
+        end
+
+        def literal_number?(node)
+          node.int_type? || node.float_type?
+        end
+
+        def pluralize(method_name)
+          DURATION_METHODS.fetch(method_name.to_sym)
+        end
+
+        def singularize(method_name)
+          DURATION_METHODS.invert.fetch(method_name.to_sym)
```

Comment:
I don't think I have ever used `invert` before. I am going to have to remember that one.

Performance wise, it seems to take significantly longer to have to `invert` the hash. It would be a good idea to extract `DURATION_METHODS.invert` to a constant.

```
INVERT = DURATION_METHODS.invert
Benchmark.ips do |x|
  x.report('regular') { DURATION_METHODS.fetch(:day) }
  x.report('invert') { DURATION_METHODS.invert.fetch(:days) }
  x.report('const') { INVERT.fetch(:days) }
  x.compare!
end  
Calculating -------------------------------------
             regular   140.330k i/100ms
              invert    34.673k i/100ms
               const   142.347k i/100ms
-------------------------------------------------
             regular      8.298M (± 4.3%) i/s -     41.397M
              invert    451.566k (±27.3%) i/s -      2.080M
               const      8.329M (± 4.3%) i/s -     41.565M

Comparison:
               const:  8328753.3 i/s
             regular:  8297891.3 i/s - 1.00x slower
              invert:   451566.4 i/s - 18.44x slower
```